Erika a des préférences données par $u(X) = \ln X$. Elle a un billet de lotterie qu'elle veut vendre. Il y a 75\% de chance qu'elle gagne 100 dollars et 25\% de chance qu'elle perdre 100 dollars. Sa richesse actuelle est de 500 dollars. Quel est le prix minimum pour lequel elle est prête à vendre le billet?  

In [5]:
import numpy as np 

In [6]:
def u(x):
  return np.log(x)

In [7]:
p = 0.75
L = [100,-100]
x0 = 500

In [18]:
z = p*L[0] + (1-p)*L[1]
z

50.0

In [8]:
def eu(p,L,x0):
  return p*u(x0 + L[0]) + (1-p)*u(x0+L[1])

In [14]:
eu_lot = eu(p,L,x0)
eu_lot

6.2955633781891045

In [11]:
def slack(prix,p,L,x0):
  return u(x0+prix) - eu(p,L,x0)

In [12]:
from scipy.optimize import bisect

In [13]:
bisect(slack,0.0,0.5*x0,args=(p,L,x0))

42.161202165905465

In [17]:
500 - np.exp(6.2955633781891045)

-42.1612021659065

À la main, 

On a $u(x) = \log x$. L'utilité espérée est de :

$$ Eu = 0.75  \log 600 + 0.25  \log 400  = 6.295$$

On pose ensuite: 

$$\log (500 + \pi) = 6.2955633781891045 $$

et donc 

$$ \pi = \exp(6.2955633781891045)-500 = 42.161 $$ 